packages/react-devtools-shared/src/hook.js

```diff
@@ -340,37 +332,24 @@ export function installHook(target: any): DevToolsHook | null {
     // * Disabling or marking logs during a double render in Strict Mode
     // * Disable logging during re-renders to inspect hooks (see inspectHooksOfFiber)
     //
-    // For React Native, we intentionally patch early (during injection).
-    // This provides React Native developers with components stacks even if they don't run DevTools.
-    //
-    // This won't work for DOM though, since this entire file is eval'ed and inserted as a script tag.
-    // In that case, we'll only patch parts of the console that are needed during the first render
-    // and patch everything else later (when the frontend attaches).
-    //
-    // Don't patch in test environments because we don't want to interfere with Jest's own console overrides.
-    //
-    // Note that because this function is inlined, this conditional check must only use static booleans.
-    // Otherwise the extension will throw with an undefined error.
-    // (See comments in the try/catch below for more context on inlining.)
-    if (!__TEST__ && !__EXTENSION__) {
-      try {
-        // The installHook() function is injected by being stringified in the browser,
-        // so imports outside of this function do not get included.
-        //
-        // Normally we could check "typeof patchConsole === 'function'",
-        // but Webpack wraps imports with an object (e.g. _backend_console__WEBPACK_IMPORTED_MODULE_0__)
-        // and the object itself will be undefined as well for the reasons mentioned above,
-        // so we use try/catch instead.
+    // Allow patching console early (during injection) to
+    // provide developers with components stacks even if they don't run DevTools.
+    if (target.hasOwnProperty('__REACT_DEVTOOLS_CONSOLE_FUNCTIONS__')) {
+      const {registerRendererWithConsole, patchConsoleUsingWindowValues} =
+        target.__REACT_DEVTOOLS_CONSOLE_FUNCTIONS__;
+      if (
+        typeof registerRendererWithConsole === 'function' &&
+        typeof patchConsoleUsingWindowValues === 'function'
+      ) {
         registerRendererWithConsole(renderer);
         patchConsoleUsingWindowValues();
-      } catch (error) {}
```

Comment:
What could error before and why are we removing the catch?

Reply:
That's a great question! In the original implementation, this file is injected into the page via an inline script tag with `installHook.toString()`. For example: https://github.com/facebook/react/blob/2b903da355fa53b3ac5a7f340134575119de2f40/packages/react-devtools-extensions/src/injectGlobalHook.js#L92-L98
Because webpack will add some prefix (like `_backend_console__WEBPACK_IMPORTED_MODULE_0__` to the modules imported in hook.js during compiling, it would actually always cause an error in the extension. This `try-catch` is for that purpose.
The comments on the top of hook.js says "we have to inline the whole ... implementation here" for the same reason.
Now we are using a much more elegant way to inject the hook, this is no longer necessary.

Reply:
People probably wanted to patch as early as possible for all platforms. But they didn't find a way to make it work for the extension. Then they noticed that the DevTools backend is initialized automatically by the extension, so it was accepted to patch it a bit later.

Reply:
This patched console depends on `WorkTagMap`, which is defined in the reconciler and can be changed in different versions. Going forward I think it's probably the best to keep it with the devtools backend. On dev and profiling builds we will be able to patch it early even on the web. On prod builds it should be OK to not have this additional information before the DevTools backend is initialized.

I'd like to hear thoughts from @gaearon and @acdlite, if any